### ISSUE TITLE
Corruption of TIFF images that was introduced in 5.0.0

### ISSUE THREAD

**Issue by alexwlchan** [python-pillow/Pillow#4621]
I’m one of the maintainers of Loris, an image server that uses Pillow. One of our users has reported corruption of TIFF images that I believe comes from Pillow (https://github.com/loris-imageserver/loris/issues/503).

This example downloads the TIFF, and converts it to a JPEG (image from https://elifesciences.org/digests/55692/could-some-antibiotics-be-immune-stimulants):

```python
import hashlib
import os
import shutil
from urllib.request import urlretrieve

from PIL import Image

# Download the image
filename, _ = urlretrieve("https://prod-elife-published.s3.amazonaws.com/digests/55692/digest-55692.tif")

# Expected size: 219528
# Expected hash: 01ddb8885aa5252b41be06dcdf36703932be24436013d5ea65ca9b1fd675258a
print(f"size = {os.stat(filename).st_size}")
print(f"sha256 = {hashlib.sha256(open(filename, 'rb').read()).hexdigest()}")

# Open the image as a TIF; save as a JPEG
im = Image.open(filename)
im.save("digest-55692.jpg")
```

This table shows the output from three versions of Pillow (resized for readability):

<table>
<tr>
<td><img src="https://user-images.githubusercontent.com/301220/81709252-a93c4e00-9469-11ea-8370-71022743b60c.jpg"></td>
<td><img src="https://user-images.githubusercontent.com/301220/81709287-afcac580-9469-11ea-8e4a-7a1d6b548c52.jpg"></td>
<td><img src="https://user-images.githubusercontent.com/301220/81709345-b6593d00-9469-11ea-8630-272d10e4576c.jpg"></td>
</tr><tr><td>4.3.0</td><td>5.0.0 (next version)</td><td>7.1.2 (latest)
</td></tr></table>

The corruption is not deterministic; running the script repeatedly gets images that look different every time.

I have attached [a zip archive (`pillow_test_cases.zip`)](https://github.com/python-pillow/Pillow/files/4616394/pillow_test_cases.zip) which includes the original TIFF, and the three JPEG outputs.

I produced these examples running inside a Docker container, built with the following Dockerfile template and varying the version of Pillow:

```dockerfile
FROM ubuntu:18.04

RUN apt-get update
RUN apt-get install -y python3 python3-pip

RUN pip3 install Pillow==5.0.0

COPY example.py /

CMD ["python3", "/example.py"]
```

We've had issues with TIFFs in Loris/Pillow in the past, which led to https://github.com/python-pillow/Pillow/issues/2926. I wonder if this issue has a similar cause?

**Comment by radarhere** [comment]
Thanks for your detailed report.

Testing, I find that the problem begins with #2899, when we removed our internal TIFF LZW decoder.

Trying to look for an unusual setup in the TIFF file, I see that it has a Planar Configuration value of 2 and LZW compression. All of the test images from Pillow that I could open with Planar Configuration 2 are raw compression, so I suspect the use of compression plus separate component planes is causing the difficulty.

**Comment by kkopachev** [comment]
Looking at TiffDecode.c, Planar Configuration = 2 is not supported at all. So all compressed tiffs would fail in that case.

**Comment by radarhere** [comment]
Should be resolved by #5364
